src/main/java/org/skife/jdbi/v2/StatementContext.java

```diff
@@ -55,6 +57,14 @@
      */
     Map<String, Object> getAttributes();
 
+    /**
+     * Obtain a column mapper for the given type in this context.
+     *
+     * @param type the target type to map to
+     * @return a ResultColumnMapper for the given type
+     * @throws org.skife.jdbi.v2.exceptions.DBIException if no mappers are registered for the given type.
+     */
+    ResultColumnMapper columnMapperFor(Class type);
```

Comment:
I see how this could be useful for automatic composition, e.g. chaining an `OptionalMapper` and a `LocalDateMapper` to read a `date` column into an `Optional<LocalDate>`.

I chose `Class` here to make it agree with existing API, in particular `ResultSetMapperFactory.mapperFor(Class,StatementContext)`.

Are you ok with making the `ResultSetMapper` and `ResultColumnMapper` interfaces diverge in v2?